lib/src/algorithms.dart

```diff
@@ -171,7 +172,7 @@ void mergeSort<T>(List<T> list,
   var firstLength = middle - start;
   var secondLength = end - middle;
   // secondLength is always the same as firstLength, or one greater.
-  var scratchSpace = List<T>(secondLength);
+  var scratchSpace = List<T?>(secondLength);
```

Comment:
I thought this constructor was going away and we'd have to use `List<T?>.filled(secondLength, null)`

Reply:
It works in the current SDK - removing that entirely would break huge swaths of existing code? I don't think we have a way to only remove it in the nnbd sdk...

Reply:
https://github.com/dart-lang/language/issues/746#issuecomment-571377956

cc @leafpetersen - is there a tracking issue for implementing the error?

Reply:
Fwiw the migrator tool also does not migrate this yet as it was indidcated it should in that issue. That would alleviate some of my concern here. cc @stereotype441

Reply:
I have updated everything to use `List.filled` now though.